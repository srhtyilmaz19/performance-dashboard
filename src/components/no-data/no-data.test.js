import React from "react";

import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import NoData from "./no-data";

afterEach(cleanup);

describe("<NoData />", () => {
  it("should render error component", function () {
    const { asFragment } = render(<NoData />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have class", function () {
    const handleClick = () => console.log("handleClick triggered !");

    const { getByTestId } = render(<NoData onClick={handleClick} />);

    const button = getByTestId("no-data-item-wrapper");

    expect(button).toHaveClass("item-wrapper");
  });

  it("should text class", function () {
    const handleClick = () => console.log("handleClick triggered !");

    const { getByTestId } = render(<NoData onClick={handleClick} />);

    const button = getByTestId("no-data-item");

    expect(button).toHaveTextContent("No valid data found.");
  });
});
