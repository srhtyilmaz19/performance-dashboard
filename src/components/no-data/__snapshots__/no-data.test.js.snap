// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoData /> should render error component 1`] = `
<DocumentFragment>
  <div
    class="item-wrapper"
    data-testid="no-data-item-wrapper"
  >
    <p
      class="no-data-text"
      data-testid="no-data-item"
    >
      No valid data found.
    </p>
  </div>
</DocumentFragment>
`;
